shark_turbine/kernel/wave/codegen.py

```diff
@@ -1,13 +1,57 @@
 import operator
-from typing import Any, Callable, ClassVar, Optional
+import sympy
+from typing import Any, Callable, ClassVar, Optional, List, Type
 from dataclasses import dataclass
 import torch.fx as fx
 
+from ..compiler.ir import (
+    InsertionPoint,
+    Location,
+    OpResult,
+    Value,
+    IndexType,
+    MemRefType,
+    VectorType,
+    IntegerAttr,
+    arith_d,
+    func_d,
+    gpu_d,
+    stream_d,
+    vector_d,
+)
+
+# TK infrastructure imports.
+from shark_turbine.kernel.lang.global_symbols import *
 from ..ops.wave_ops import write, register, mma, read, reduction
-from ..compiler.base import CodegenError
-from ..compiler.ir import InsertionPoint, Location
+from ..compiler.base import CodegenError, NDEBUG
 from ..compiler.kernel_codegen import BoundKernelSignature
 from .._support.tracing import CapturedTrace
+from ..compiler.builder import ScalarBuilder, IRProxyValue
+from ..compiler.vector_codegen import (
+    cast_kernel_buffer,
+    cast_py_literal,
+    cast_py_value,
+    cast_vector,
+)
+
+# Indexing imports.
+from .._support.indexing import IndexingContext
+
+
+@dataclass
+class NodeAttrs:
+    # By default, integers are assumed signed. We propagate unsigned as graph
+    # node attrs.
+    unsigned: bool = False
+
+    @staticmethod
+    def load(py_value) -> "NodeAttrs":
+        if isinstance(py_value, fx.Node):
+            return NodeAttrs(unsigned=bool(py_value.meta.get("unsigned")))
+        return NodeAttrs()
+
+    def store(self, node: fx.Node):
```

Comment:
Also instead of using the fx.Node here and below, can we use the CustomNode? The reason is that the CustomNode contains more information that is easily queryable as opposed to the fx.node.